zcash_primitives/src/transaction/mod.rs

```diff
@@ -44,8 +44,8 @@ impl fmt::Display for TxId {
 /// A Zcash transaction.
 #[derive(Debug)]
 pub struct Transaction {
-    txid: TxId,
-    data: TransactionData,
+    pub txid: TxId,
+    pub data: TransactionData,
```

Comment:
Why are you making this change? Providing public access to `TransactionData` with the current `Transaction` design undermines the type-system enforcement that the cached `txid` is correct for `data`, because anyone can alter the internals.

Reply:
I'm accessing the `.data` field directly in  a few places in the tests, such as https://github.com/str4d/librustzcash/pull/12/files#diff-5864f04538849ea3123c13f45a0ba314R673 - what do you think, should I add a trait for access to these members?